Libclang/symbols.test fails on Linux when using the `mold` linker

I'm unsure if this is a mold issue or a clang issue, so I'm filing this bug here first.

Repro steps:

I have LLVM commit a7a78fd427569a7ad8a27e682a66fe414f004a35 checked out.

```
$ clang --version
Debian clang version 16.0.6 (16)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
$ mold --version
mold 2.2.0 (compatible with GNU ld)
$ cmake -DLLVM_USE_LINKER=mold  -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DLLVM_TARGETS_TO_BUILD='X86' -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_PROJECTS='clang' -G Ninja ../llvm
```

Failure message:

```
$ ninja check-clang
[3527/3528] Running the Clang regression tests
llvm-lit: /usr/local/google/home/ayzhao/src/llvm-project/llvm/utils/lit/lit/llvm/config.py:488: note: using clang: /usr/local/google/home/ayzhao/src/llvm-project/build-mold/bin/clang
FAIL: Clang :: LibClang/symbols.test (11870 of 20499)
******************** TEST 'Clang :: LibClang/symbols.test' FAILED ********************
Exit Code: 1

Command Output (stdout):
--
00000000010e3e70 t clang_Cursor_getTranslationUnit$plt
00000000010e3e60 t clang_Cursor_isNull$plt
00000000010e3eb0 t clang_Range_isNull$plt
00000000039414c8 d clang_createTranslationUnit$got
00000000010e3e40 t clang_defaultDiagnosticDisplayOptions$plt
00000000010e3f10 t clang_disposeString$plt
00000000010e3e00 t clang_equalCursors$plt
00000000010e3e30 t clang_formatDiagnostic$plt
00000000010e3f00 t clang_getCString$plt
00000000010e3db0 t clang_getCursorKind$plt
00000000010e3df0 t clang_getCursorLocation$plt
00000000010e3e10 t clang_getCursorReferenced$plt
00000000010e3e20 t clang_getCursorSemanticParent$plt
00000000010e3e50 t clang_getCursorUSR$plt
00000000010e3ef0 t clang_getFileLocation$plt
00000000010e3dd0 t clang_getFileName$plt
00000000010e3dc0 t clang_getNullCursor$plt
00000000010e3e80 t clang_getNullLocation$plt
00000000010e3e90 t clang_getNullRange$plt
00000000010e3ea0 t clang_getRange$plt
00000000010e3ed0 t clang_getRangeEnd$plt
00000000010e3ec0 t clang_getRangeStart$plt
00000000010e3ee0 t clang_getSpellingLocation$plt
00000000010e3f20 t clang_getTypeDeclaration$plt
00000000010e3f30 t clang_install_aborting_llvm_fatal_error_handler$plt
00000000010e3d80 t clang_isDeclaration$plt
00000000010e3da0 t clang_isExpression$plt
00000000010e3d70 t clang_isInvalid$plt
00000000010e3d90 t clang_isReference$plt
00000000010e3de0 t clang_visitChildren$plt

--
Command Output (stderr):
--
RUN: at line 2: llvm-nm -Dj --defined-only /usr/local/google/home/ayzhao/src/llvm-project/build-mold/lib/libclang.so | grep -v -e '@@LLVM_[0-9]\+$' | /usr/local/google/home/ayzhao/src/llvm-project/build-mold/bin/not grep '^clang'
+ grep -v -e '@@LLVM_[0-9]\+$'
+ /usr/local/google/home/ayzhao/src/llvm-project/build-mold/bin/not grep '^clang'
+ llvm-nm -Dj --defined-only /usr/local/google/home/ayzhao/src/llvm-project/build-mold/lib/libclang.so
RUN: at line 6: llvm-nm /usr/local/google/home/ayzhao/src/llvm-project/build-mold/lib/libclang.so | /usr/local/google/home/ayzhao/src/llvm-project/build-mold/bin/not grep '[a-z] clang_[^.]\+$'
+ llvm-nm /usr/local/google/home/ayzhao/src/llvm-project/build-mold/lib/libclang.so
+ /usr/local/google/home/ayzhao/src/llvm-project/build-mold/bin/not grep '[a-z] clang_[^.]\+$'

--

********************
********************
Failed Tests (1):
  Clang :: LibClang/symbols.test


Testing Time: 33.40s

Total Discovered Tests: 35168
  Skipped          :    35 (0.10%)
  Unsupported      :  3411 (9.70%)
  Passed           : 31697 (90.13%)
  Expectedly Failed:    24 (0.07%)
  Failed           :     1 (0.00%)
FAILED: tools/clang/test/CMakeFiles/check-clang /usr/local/google/home/ayzhao/src/llvm-project/build-mold/tools/clang/test/CMakeFiles/check-clang
cd /usr/local/google/home/ayzhao/src/llvm-project/build-mold/tools/clang/test && /usr/bin/python3 /usr/local/google/home/ayzhao/src/llvm-project/build-mold/./bin/llvm-lit -sv --param USE_Z3_SOLVER=0 /usr/local/google/home/ayzhao/src/llvm-project/build-mold/tools/clang/test
ninja: build stopped: subcommand failed.
```

--- a/clang/test/LibClang/symbols.test
+++ b/clang/test/LibClang/symbols.test
@@ -1,6 +1,6 @@
 # Check that there are no unversioned clang symbols in libclang.so
 RUN: llvm-nm -Dj --defined-only %libclang | grep -v -e '@@LLVM_[0-9]\+$' | not grep '^clang'
 
-# Check that here are no local clang_ symbols (ignoring symbols with .cold or
+# Check that there are no local clang_ symbols (ignoring symbols with .cold or
 # .localalias suffxies.)
-RUN: llvm-nm %libclang | not grep '[a-z] clang_[^.]\+$'
+RUN: llvm-nm %libclang | not grep '[a-z] clang_[^.$]\+$'

